### ISSUE TITLE
Question - view idea - sort by group, explicit-label, name

### ISSUE THREAD

**Issue by smallscript** [koalamer/vsc-labeled-bookmarks#3]
First,
I **really appreciate** all the work and the design. I have found setting EMOJIs 🦜 and using groups to be great.

Setting a LABEL on a bookmark and STILL seeing its bookmark-selection-text is GREAT. (i.e., using ✏️ on a TV Bookmark entry)
Seeing the BOXED LABEL in the code is GREAT.

It is so nice that honestly I am happily frustrated with wanting an alternate group view that collates by
- group (icon; group-name; placing the focused group at the top)
- then subgroup by BOOMARK's Explicit LABEL; or FILENAME if NONE (right now groups by file only; file level ordering ???)
- then by SELECTIONTXT-FILE-LINE-# (although it would be GREAT to be able to manually reorder at this level)

I would love to have that group view option.

Again, thank you so much for your work on this plugin!!!

**Comment by koalamer** [comment]
Hello!

I'm glad you like it!

As a foreword to the new view idea: to get the tree view working acceptably was a bit of a pain in the ace, so I'm cautiously enthusiastic about another one or about touching the current one with a stick, even.

The idea to mix labels and file names in the ordering sounds like a very special kind of ordering, I don't think it would be practical for most users. Or at least it sounds confusing to me, when I think about it.
The current ordering BTW is that groups are ordered by name and files are ordered by their project relative path. (And of course bookmarks are further ordered by the line number and the character number where the bookmark was placed, though the character number is not displayed anywhere.)
Originally I wanted to display folder levels to make files better identifiable, but the API gives no help with that, I would have to fake those levels myself, so I dropped the idea.
So all in all, I would rather not do this kind of ordering.

Manual ordering would require a duplicate list of bookmarks, which sounds nasty, when it comes to meaningfully placing new items into the list. Furthermore, the tree view API does not offer drag and drop operations, so you could at best do ordering by entering some weight parameter for each item in an input field, one by one...
So the manual ordering could only be done in a user unfriendly way. I don't think it's worth it without drag and drop capabilities. If drag and drop becomes available, then maybe.

Sorting bookmarks by the label first and then by the usual location based stuff could kind of make sense, but there is a technical dependency: it would require that alternatively ordered bookmark list is created and maintained.
The bookmark list can change on every edit event, which means on every keystroke, and that might make maintenance of a secondary list processing heavy. 
So this one would be maybe doable but would require effort.

Putting the active group first might be easily doable though. I could try make that an option, if you think that's helpful.
Or I could mark the active group somehow in the tree view. (I originally thought about using the dimmed markers for inactive groups there too, but that is the only place where they can shine whether active or not, so I dropped that idea.)

Basically, sorting is bound to how the core list is ordered: by location. The tree view and the navigation lists are the way they are not because of some crazy design decision, but because that's how they could be done without introducing additional complexity.

So, maybe not the kind of answer you hoped for, but there it is... I would consider doing something about the active group display in the tree, but not the rest. An even for that I ask for your patience as this is a side project of mine, where deadlines do not really exist.

**Comment by koalamer** [comment]
Hello!

It has been a while and though I rejected your suggestions, there is an update coming that might be somewhat interesting to you: I've split the 'by group' tree view into two, one for the active group only, and the other for the inactive ones.

The groups are ordered by name, and the files by their project relative paths, as always.

You can test drive it if you wish: https://www.dropbox.com/s/lwei23iz9zr3ndz/labeled-bookmarks-1.1.10.vsix?dl=0
Unless bugs are found, this will be the next release.

With that, I'm closing this issue. Sorry I could not fulfill your requests, but they just don't fit the system.
